DIL ACM/read_int.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+// takes the token. Returns the base, or 0 if error
+int base_of(const char *);
+
+// takes ONE character and the base. returns -1 on error
+int read_int(char, const int);
```

Comment:
These functions are not supposed to be used by the user and hence should be "private", i.e. `static` in the file they're used.

Reply:
Shall I declare them in the cpp file? Is that what you mean?

Reply:
I mean first of all to remove them from here. In the `.h` files you put functions that are supposed to be used by the user. To implement something, you may use _more_ functions than those that are supposed to be used by the user. For example here, the user is supposed to only call (and care) for `read_int(const char *, int *);`. But, to implement this function, you have defined other functions, that the user does not need to know about. So, you don't declare them here.
What you do is you _define_ them in the cpp file. You don't necessarily need to declare them if they're defined before they're used (tell if you're confused about the difference of a definition and a declaration).
Now, the extra thing that you do is that you define the function `static`, e.g.:
`static int base_of(const char *);`. In that way, this function can't be called from another source file (this is a very simplistic and almost incorrect description, but I don't want to make it complicated at this point :) ).